cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java

```diff
@@ -1277,6 +1290,13 @@ private void sanityCheckGoalNames() {
         throw new ConfigException("Attempt to configure goals with case sensitive names.");
       }
     }
+    // Check goal names are case insensitive for intra-broker goals.
+    for (String goalName: getList(KafkaCruiseControlConfig.INTRA_BROKER_GOALS_CONFIG)) {
+      if (!caseInsensitiveGoalNames.add(goalName.replaceAll(".*\\.", ""))) {
+        throw new ConfigException("Attempt to configure intra.broker.goals with case sensitive names.");
```

Comment:
Nit: If we intend to
(1) use exact config name, can we use `INTRA_BROKER_GOALS_CONFIG` in a formatted string?
(2) not use exact config name, would it be possible to change `intra.broker.goals` to `intra-broker goals`?